src/api/include/migraphx/migraphx.hpp

```diff
@@ -979,6 +993,18 @@ struct program : MIGRAPHX_HANDLE_BASE(program)
         return arguments(pout, own{});
     }
 
+    /// Overloaded to allow for excecuction_environment input
+    arguments eval(const program_parameters& pparams, const execution_environment& e) const
```

Comment:
This should be called `run_async`. Ideally, I would like to deprecate `eval` and replace it with `run` so it matches the python API.

Reply:
That's fine. So I'll just rename this and change the call chain appropriately. while leaving the old eval() intact till we deprecate it then?

Reply:
done. Made this run_async, and put back eval() to the wait it was before these changes. If we want to rename this to run() to be consistent with the API I can do that as part of this changeset too.